### PR TITLE
[Bugfix] Add  fully sharded layer for QKVParallelLinearWithLora

### PR DESCRIPTION
**Pull request by jeejeelee**
FILL IN THE PR DESCRIPTION HERE

Due to the lack of a fully sharded layer for `QKVParallelLinearWithLora`, the current fully sharded loras cannot support certain LLMs, such as `Baichuan-7B`.

 Reproduce  snippet:
```python
def test_baichuan_lora(baichuan_lora_files):
    MODEL_PATH = "baichuan-inc/Baichuan-7B"
    llm = vllm.LLM(
        MODEL_PATH,
        max_model_len=1024,
        enable_lora=True,
        max_loras=4,
        max_lora_rank=64,
        trust_remote_code=True,
        gpu_memory_utilization=0.5,
        tensor_parallel_size=2,
        fully_sharded_loras=True,
    )
    expected_lora_output = [
        "SELECT count(*) FROM singer",
        "SELECT avg(age) ,  min(age) ,  max(age) FROM singer WHERE Country  =  'France'",  # noqa: E501
        "SELECT name ,  country ,  age FROM singer ORDER BY age ASC",
    ]

    output1 = do_sample(llm, baichuan_lora_files, lora_id=1)
    for i in range(len(expected_lora_output)):
        assert output1[i] == expected_lora_output[i]
    output2 = do_sample(llm, baichuan_lora_files, lora_id=2)
    for i in range(len(expected_lora_output)):
        assert output2[i] == expected_lora_output[i]
if __name__ == "__main__":
    lora_path = snapshot_download(repo_id="jeeejeee/baichuan7b-text2sql-spider")
    test_baichuan_lora(lora_path)
```

ping @Yard1

### CODE REVIEW

File: tests/lora/test_baichuan.py [hunk 1]
```diff
@@ -81,13 +81,16 @@ def test_baichuan_tensor_parallel_equality(baichuan_lora_files):
...
-                       trust_remote_code=True)
+    llm_tp2 = vllm.LLM(
+        MODEL_PATH,
+        enable_lora=True,
+        max_num_seqs=16,
+        max_loras=4,
+        max_lora_rank=64,
+        tensor_parallel_size=2,
+        trust_remote_code=True,
+        fully_sharded_loras=True,
```
Comment (Yard1) [inline tests/lora/test_baichuan.py:92]: can we parametrize this and test both fully sharded True and False?

**Review by Yard1 — COMMENTED** [review]
LGTM, just one nit

File: tests/lora/test_baichuan.py (hunk 1 above)
Comment (jeejeelee) [reply tests/lora/test_baichuan.py]: Done， please review again, thank you.

File: examples/offline_inference.py [hunk 2]
```diff
@@ -19,4 +19,4 @@
 for output in outputs:
     prompt = output.prompt
     generated_text = output.outputs[0].text
-    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
+    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
```
Comment (Yard1) [inline examples/offline_inference.py:22]:
```suggestion
    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")

```

**Review by Yard1 — APPROVED** [review]

**Pull request merged**